File: send_test_email.py
#!/usr/bin/env python3
import os
import sys
import requests
from dotenv import load_dotenv

# Load environment variables from .env.secrets
load_dotenv('.env.secrets')

def send_simple_message():
    api_key = os.getenv('MAILGUN_API')
    
    if not api_key:
        print("Error: MAILGUN_API not found in .env.secrets")
        return False
    
    try:
        response = requests.post(
            "https://api.mailgun.net/v3/sandboxad2ef991cd4643e990f0338482f1cb90.mailgun.org/messages",
            auth=("api", api_key),
            data={
                "from": "Mailgun Sandbox <postmaster@sandboxad2ef991cd4643e990f0338482f1cb90.mailgun.org>",
                "to": "tianzhi chen <burgesschen1990@gmail.com>",
                "subject": "Hello tianzhi chen",
                "text": "Congratulations tianzhi chen, you just sent an email with Mailgun! You are truly awesome!"
            }
        )
        
        if response.status_code == 200:
            print("Email sent successfully!")
            print(f"Response: {response.json()}")
            return True
        else:
            print(f"Failed to send email. Status code: {response.status_code}")
            print(f"Response: {response.text}")
            return False
            
    except Exception as e:
        print(f"Error sending email: {str(e)}")
        return False

if __name__ == "__main__":
    success = send_simple_message()
    sys.exit(0 if success else 1)

File: SendTestEmail.java
import java.io.File;
import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.JsonNode;
import com.mashape.unirest.http.Unirest;
import com.mashape.unirest.http.exceptions.UnirestException;

public class SendTestEmail {
    public static void main(String[] args) {
        try {
            JsonNode result = sendSimpleMessage();
            System.out.println("Email sent successfully!");
            System.out.println("Response: " + result.toString());
        } catch (UnirestException e) {
            System.err.println("Failed to send email: " + e.getMessage());
            System.exit(1);
        }
    }
    
    public static JsonNode sendSimpleMessage() throws UnirestException {
        String apiKey = System.getenv("API_KEY");
        if (apiKey == null) {
            apiKey = "API_KEY";
        }

        HttpResponse<JsonNode> request = Unirest.post("https://api.mailgun.net/v3/sandbox40099185e5954b6c884f2ae7bc6c5a72.mailgun.org/messages")
            .basicAuth("api", apiKey)
            .queryString("from", "Mailgun Sandbox <postmaster@sandbox40099185e5954b6c884f2ae7bc6c5a72.mailgun.org>")
            .queryString("to", "tianzhi chen <tianzhic.dev@gmail.com>")
            .queryString("subject", "Hello tianzhi chen")
            .queryString("text", "Congratulations tianzhi chen, you just sent an email with Mailgun! You are truly awesome!")
            .asJson();
        return request.getBody();
    }
}